src/snowflake/connector/connection.py

```diff
@@ -349,6 +365,17 @@ def __init__(
                     f" known ones are {list(connections.keys())}"
                 )
             kwargs = {**connections[connection_name], **kwargs}
+        elif empty_kwargs:
+            # connection_name is None and kwargs was empty when called
+            def_connection_name = CONFIG_MANAGER["default_connection_name"]
```

Comment:
Do we expect "default_connection_name" a must in the config file? If no, this may raise an KeyError. Suggest we raise a meaningful error message to users.

Reply:
It's not, it has a default value

Reply:
Just to be clear, it's not possible for this to ever be a `KeyError`?

Reply:
Based on my testing it's not